code/modules/mob/floating_message.dm

```diff
@@ -14,27 +19,45 @@ var/list/floating_chat_colors = list()
 
 	var/static/regex/html_metachars = new(@"&[A-Za-z]{1,7};", "g")
 	message = replacetext(message, html_metachars, "")
-
+	var/expression = STATING
 	var/style	//additional style params for the message
 	var/fontsize = 4
-	if(small)
-		fontsize = 2
+	var/copied_text_ending = copytext_char(message, -1)
 	var/limit = 160
-	if(copytext_char(message, length_char(message) - 1) == "!!")
-		fontsize = 8
-		limit = 160
-		style += "font-weight: bold;"
-
+	if(small)
+		expression |= WHISPERING
+	if(copied_text_ending == "!")
+		expression |= YELLING
+	if(copied_text_ending == "?")
+		expression |= ASKING
 	if(length_char(message) > limit)
 		message = "[copytext_char(message, 1, limit)]..."
 
 	if(!floating_chat_colors[src])
 		floating_chat_colors[src] = get_random_colour(0,160,230)
-	style += "color: [floating_chat_colors[src]];"
+	style += "color: #[floating_chat_colors[src]]; "
 
 	// create 2 messages, one that appears if you know the language, and one that appears when you don't know the language
-	var/image/understood = generate_floating_text(src, capitalize(message), style, fontsize, duration, show_to)
-	var/image/gibberish = speaking ? generate_floating_text(src, speaking.scramble(message), style, fontsize, duration, show_to) : understood
+	var/image/understood = generate_floating_text(src, \
+		capitalize(message), \
+		style, \
+		fontsize, \
+		duration, \
+		show_to, \
+		expression, \
+		colour = floating_chat_colors[src]
+	)
+	var/image/gibberish = speaking ? \
+		generate_floating_text(src, \
+		speaking.scramble(message), \
+		style, \
+		fontsize, \
+		duration, \
+		show_to, \
+		expression, \
+		colour = floating_chat_colors[src]
+	) \
+	: understood
```

Comment:
Formatting Suggestion:
```suggestion
	var/image/understood = generate_floating_text(
		src,
		capitalize(message),
		style,
		fontsize,
		duration,
		show_to,
		expression,
		colour = floating_chat_colors[src],
	)
	var/image/gibberish = speaking ? generate_floating_text(
		src,
		speaking.scramble(message),
		style,
		fontsize,
		duration,
		show_to,
		expression,
		colour = floating_chat_colors[src],
	) : understood
```

Reply:
> Formatting Suggestion:
> 
> Reasoning for `if` over `?:`, if a ternary extends more than a line, readability suffers. And no there isn't a noticeable cost difference for `()?{}:{}` vs `if(){}else{}`
> ```suggestion
> 	var/image/understood = generate_floating_text(
> 		holder = src,
> 		message = capitalize(message),
> 		style = style,
> 		size = fontsize,
> 		duration = duration,
> 		show_to = show_to,
> 		expression = expression,
> 		colour = floating_chat_colors[src],
> 	)
> 	var/image/gibberish
> 	if(speaking)
> 		gibberish = generate_floating_text(
> 			holder = src,
> 			message = speaking.scramble(message),
> 			style = style,
> 			size = fontsize,
> 			duration = duration,
> 			show_to = show_to,
> 			expression = expression,
> 			colour = floating_chat_colors[src],
> 		)
> 	else
> 		gibberish = understood
> ```

The check is within the variable assignment, i don't think you can embed and if statement in those.

Reply:
You can move the variable assignment into the if.